src/aws/roles.tf

```diff
@@ -214,6 +216,51 @@ resource "aws_iam_role_policy_attachment" "media_convert_s3_policy_attachment" {
   policy_arn = aws_iam_policy.media_convert_s3_policy.arn
 }
 
+resource "aws_iam_role" "convert_lambda_invocation_role" {
+  name = "${terraform.workspace}-marsha-convert-lambda-invocation-role"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "lambda.amazonaws.com"
+      },
+      "Effect": "Allow"
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_policy" "convert_lambda_invocation_policy" {
+  name        = "${terraform.workspace}-marsha-convert-lambda-invocation-policy"
+  path        = "/"
+  description = "IAM policy for invoking convert lambda"
+
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": [
+        "lambda:InvokeFunction"
+      ],
+      "Resource": "${aws_lambda_function.marsha_convert_lambda.arn}",
+      "Effect": "Allow"
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy_attachment" "media_convert_invocation_policy_attachment" {
+  role       = aws_iam_role.convert_lambda_invocation_role.name
```

Comment:
you attache the policy to a role but what are you doing of this role ? The user created by terraform is still not allowed to invoke the function.
`aws_iam_role` and `aws_iam_policy` are not the resources to use to allow a user to do something on AWS.

You should take a look at the `aws_iam_user_policy` instead: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/iam_user_policy
We are already using it in `user.tf` file.